node/test/retry.js

```diff
@@ -27,42 +27,34 @@ var TChannel = require('../channel');
 allocCluster.test('request retries', {
     numPeers: 4
 }, function t(cluster, assert) {
-    var random = randSeq([
-        1.0, 0.1, 0.1, 0.1, // .request, chan 1 wins
-        1.0,                // chan 1 declines
-             1.0, 0.1, 0.1, // .request, chan 2 wins (1 is skipped)
-        0.25,               // chan 2 too busy
-                  1.0, 0.1, // .request, chan 3 wins (1-2 are skipped)
-        0.1,                // chan 3 unexpected error
-                       0.1, // .request, chan 4 wins (only one left)
-        0.0                 // success!
-    ]);
-
-    cluster.channels.forEach(function each(server, i) {
-        var n = i + 1;
-        var chan = server.makeSubChannel({
-            serviceName: 'tristan'
-        });
-        chan.register('foo', function foo(req, res, arg2, arg3) {
-            var rand = random();
-            if (rand >= 0.5) {
-                res.sendError('Declined', 'magic 8-ball says no');
-            } else if (rand >= 0.25) {
-                res.sendError('Busy', "can't talk");
-            } else if (rand) {
-                res.sendError('UnexpectedError', 'wat');
-            } else {
-                var str = String(arg3);
-                str = str.toUpperCase();
-                res.headers.as = 'raw';
-                res.sendOk('served by ' + n, str);
-            }
-        });
-    });
+    // chan 1 declines
+    cluster.channels[0]
+        .makeSubChannel({serviceName: 'tristan'})
+        .register('foo', declineFoo);
+
+    // chan 2 too busy
+    cluster.channels[1]
+        .makeSubChannel({serviceName: 'tristan'})
+        .register('foo', busyFoo);
+
+    // chan 3 unexpected error
+    cluster.channels[2]
+        .makeSubChannel({serviceName: 'tristan'})
+        .register('foo', unexpectedErrorFoo);
+
+    // success!
+    cluster.channels[3]
+        .makeSubChannel({serviceName: 'tristan'})
+        .register('foo', servedByFoo(4));
 
     var client = TChannel({
         timeoutFuzz: 0,
-        random: random
+        random: randSeq([
+            1.0, 0.1, 0.1, 0.1, // .request, chan 1 wins
+                 1.0, 0.1, 0.1, // .request, chan 2 wins (1 is skipped)
+                      1.0, 0.1, // .request, chan 3 wins (1-2 are skipped)
+                           0.1  // .request, chan 4 wins (only one left)
```

Comment:
2.0 would be an invalid value for a random function to return, hence the constant